packages/commands/measure/src/server/ServerApp.tsx

```diff
@@ -40,31 +40,40 @@ const useCleanupOnManualExit = () => {
   });
 };
 
-export const ServerApp = () => {
+interface ServerAppProps {
+  port: number;
+}
+
+export const ServerApp = ({ port }: ServerAppProps) => {
   const [socket, setSocket] = useState<SocketType | null>(null);
+  const webAppUrl = getWebAppUrl(port);
   useEffect(() => {
     const app = createExpressApp();
 
     const server = http.createServer(app);
     const io: SocketServer = new Server(server, {
       cors: {
-        origin: [WEBAPP_URL],
+        origin: [webAppUrl],
         methods: ["GET", "POST"],
       },
     });
 
     allowOnlyOneSocketClient(io, setSocket);
 
-    server.listen(PORT, () => {
-      open(WEBAPP_URL);
+    server.listen(port, () => {
+      open(webAppUrl);
     });
 
     return () => {
       server.close();
       io.close();
     };
-  }, []);
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
is there an issue with adding `port` and `webAppUrl` in the dependencies? 🤔

Reply:
leaving it out just ensures this effect only runs on mount. the `port` prop doesn't change, so it shouldn't be an issue either way. however if `port` and `webAppUrl` end up becoming more dynamic somehow later on, then having them as dependencies would re-run this effect on change, which could lead to unexpected behavior.

Reply:
I'd rather not disable the rule 
If `port` and `webAppUrl` end up being more dynamic, the hook would actually not work at all anyway without having them in the dependencies
(I'll admit I don't think the code will work chaging port on the fly even if we have them, just saying not having them isn't necessarily better)

I usually don't like disabling it (I try to avoid disabling eslint rule in general), because you might miss some dependencies you really do want to have in there

Reply:
> If port and webAppUrl end up being more dynamic, the hook would actually not work at all anyway without having them in the dependencies

to be clear, the hook still operates just fine without `port` and `webAppUrl` in its dependencies.

however, it won't automatically re-run if either of these values changes dynamically. including them as dependencies does ensure that the effect is re-triggered with updated values, but this comes with it's own set of challenges.

> just saying not having them isn't necessarily better

specifically, it introduces the risk of inadvertently reinitializing the server or causing memory leaks if we don't manage the lifecycle of these resources meticulously. therefore, i believe that not adding them as dependencies is a safer approach.

> because you might miss some dependencies you really do want to have in there

this way, if a dynamic value fails to update as expected, it becomes a clear indicator for other devs to investigate. in essence, avoiding these dependencies helps maintain stability and predictability in the behavior of the server setup.

Reply:
I'm not sure I agree with that but I don't think it has a big impact in this case, so I'll merge!

Thanks for the PR! Expect the new Flashlight CLI to come soon 🤞